src/GroupMap.php

```diff
@@ -77,7 +84,8 @@ public function getGroupsForEntityType($entity_type_id) {
    * @return array
    */
   public function getAllGroupBundles($entity_type = NULL) {
-    return !empty($this->groupMap[$entity_type]) ? $this->groupMap[$entity_type] : $this->groupMap;
+    $group_map = $this->groupMap;
+    return !empty($group_map[$entity_type]) ? $group_map[$entity_type] : $group_map;
```

Comment:
While your changing these can you fix the logic, for me that will lead to some WTF's. Only return all if the $entity_type is empty, and rename that to $entity_type_id.